text/0000-build-std/4-stage-1a.md

```diff
@@ -0,0 +1,1003 @@
+build-std mechanism while still being useful for users of tier three targets. By
+leaving an automatic mechanism for a later stage, fewer of the technical
+challenges of build-std need to be addressed all at once.
+
+Having an opt-in mechanism initially, such as `build-std = "always"`, allows for
+early issues with build-std to be ironed out without potentially affecting more
+users like an automatic mechanism.
+
+*See [Stage 2][stage2] for the introduction of an automatic build-std
+mechanism.*
+
+↩ [*Proposal*][proposal]
+
+## Why does "always" rebuild in release profile?
+[rationale-release-profile]: #why-does-always-rebuild-in-release-profile
+
+The release profile most closely matches the existing pre-built standard
+library, which has proven itself suitable for a majority of use cases.
+
+By minimising the differences between a newly-built std and a pre-built std,
+there is less chance of the user experiencing bugs or unexpected behaviour from
+the well-tested and supported pre-built std.
+
+*See [Stage 3][stage3] for the introduction of customised standard library
+builds.*
+
+↩ [*Proposal*][proposal]
+
+## Why add `build-std-crate`?
+[rationale-build-std-crate]: #why-add-build-std-crate
+
+Not all standard library crates will build on all targets. In a `no_std` project
+for a tier three target, `build-std-crate` gives the user the ability to limit
+which crates are built to those they know they need and will build successfully.
+
+*See [Stage 1b][stage1b] for an alternative to `build-std-crate`.*
+
+↩ [*Proposal*][proposal]
+
+## Why use the lockfile of the `rust-src` component?
+[rationale-lockfile]: #why-use-the-lockfile-of-the-rust-src-component
+
+Using different dependency versions for the standard library would invalidate
+the upstream testing of the standard library. In particular, some crates use
+unstable APIs when included as a dependency of the standard library meaning that
+there is a high risk of build breakage if any package version is changed.
+
+Using the lockfile included in the `rust-src` component guarantees that the same
+dependency versions are used as in the pre-built standard library. As the
+standard library does not re-export types from its dependencies, this will not
+affect interoperability with the same dependencies of different versions used by
+the user's crate.
+
+Using the lockfile does prevent Cargo from resolving the standard library
+dependencies to newer patch versions that may contain security fixes. However,
+this is already impossible with the pre-built standard library.
+
+See
+[*Why vendor the standard library's dependencies?*][rationale-vendoring]
+
+↩ [*Proposal*][proposal]
+
+### Why not build the standard library in incremental?
+[rationale-incremental]: #why-not-build-the-standard-library-in-incremental
+
+As the standard library sources are never modified, incremental compilation
+would only add a compilation time overhead.
```

Comment:
We could probably leave a note that this aligns with all other immutable sources (registry, git)

Reply:
Clarified in https://github.com/davidtwco/rfcs/pull/2/commits/ff37ab8779deea23f10c5dd3b458cc9976cab568